labview/labview_rabbitmq.c

```diff
@@ -5,251 +5,170 @@
+	int const FRAME_MAX = 131072;	// the maximum size of an AMQP frame. 131072 is the default. 
+									// 4096 is the minimum size, 2^31-1 is the maximum,
+									// a good default is 131072 (128KB),
+	int const HEARTBEAT = 0;	// the number of seconds between heartbeat frames to request
+								// of the broker. A value of 0 disables heartbeats.
+
+	return lv_report_amqp_error(amqp_login(conn, VHOST, CHANNEL_MAX, FRAME_MAX, HEARTBEAT, AMQP_SASL_METHOD_PLAIN, username, password), "Logging in", errorDescription);
 }
 
-
 LABVIEW_PUBLIC_FUNCTION
-int lv_amqp_create_queue(int64_t conn_intptr, uint16_t  channel, char *exchange, char *bindingkey, LStrHandle error_description) {
-	amqp_connection_state_t conn = (amqp_connection_state_t)conn_intptr;
+int lv_amqp_create_queue(int64_t conn_intptr, uint16_t channel, char* queue_name_in, LStrHandle queue_name_out, uint8_t passive, LStrHandle errorDescription)
+{
+	amqp_connection_state_t conn = (amqp_connection_state_t) conn_intptr;
 	int status;
 	amqp_bytes_t queuename;
 
-    amqp_boolean_t PASSIVE = 0;
+	if (queue_name_in!=NULL){
+		queuename = amqp_cstring_bytes(queue_name_in);
+	} else {
+		queuename = amqp_empty_bytes;
+	}
+
+	amqp_boolean_t PASSIVE = 1;
 	amqp_boolean_t DURABLE = 0;
 	amqp_boolean_t EXCLUSIVE = 0;
 	amqp_boolean_t AUTO_DELETE = 1;
-	amqp_queue_declare_ok_t *r = amqp_queue_declare(conn, channel, amqp_empty_bytes, PASSIVE, DURABLE, EXCLUSIVE, AUTO_DELETE, amqp_empty_table);
+	amqp_queue_declare_ok_t *r = amqp_queue_declare(conn, channel, queuename, PASSIVE, DURABLE, EXCLUSIVE, AUTO_DELETE, amqp_empty_table);
 
-    status = lv_report_amqp_error(amqp_get_rpc_reply(conn), "Declaring queue", error_description);
-	if (status != 1) {
-		return status;
-	}
-	
-	queuename = amqp_bytes_malloc_dup(r->queue);
-	if (queuename.bytes == NULL) {
-		copyStringToLStrHandle("Out of memory while copying queue name", error_description);
-		return _OUT_OF_MEMORY;
+	status = lv_report_amqp_error(amqp_get_rpc_reply(conn), "Declaring queue", errorDescription);
+	if (status==1){
+		copyBufferToLStrHandle(r->queue.bytes, r->queue.len, queue_name_out);
 	}
+	return status;
+}
 
-	amqp_queue_bind(conn, channel, queuename, amqp_cstring_bytes(exchange), amqp_cstring_bytes(bindingkey), amqp_empty_table);
-	status = lv_report_amqp_error(amqp_get_rpc_reply(conn), "Binding queue", error_description);
-	if (status != 1) {
+LABVIEW_PUBLIC_FUNCTION
+int lv_amqp_bind_queue(int64_t conn_intptr, uint16_t channel, char *exchange, char *queuename, char *bindingkey, LStrHandle errorDescription)
+{
+	amqp_connection_state_t conn = (amqp_connection_state_t) conn_intptr;
+	int status;
+
+	amqp_queue_bind(conn, channel, amqp_cstring_bytes(queuename), amqp_cstring_bytes(exchange), amqp_cstring_bytes(bindingkey), amqp_empty_table);
+	status = lv_report_amqp_error(amqp_get_rpc_reply(conn), "Binding queue", errorDescription);
+	if (status != 1)
+	{
 		return status;
 	}
 
 	amqp_boolean_t NO_LOCAL = 0;
```

Comment:
can you add comment to explain why these values have been chosen

Reply:
Those are default values that are set in the examples. If you need to check why - you should go to the amqp documentation and check what the particular flag is doing. I don't think it makes sense to rewrite the documentation to the code.